lib/Target/AMDGPU/AMDGPUPrintfRuntimeBinding.cpp

```diff
@@ -179,7 +177,7 @@ bool AMDGPUPrintfRuntimeBinding::shouldPrintAsStr(char Specifier,
   if (!PT) {
     return false;
   }
-  if (PT->getAddressSpace() != AMDGPUASI.CONSTANT_ADDRESS) {
+  if (PT->getAddressSpace() != 4) {
```

Comment:
Is it safe to hard code to 4?

Reply:
This change was made upstream, in the amd-common branch.  I don't really like the hard coded value either.  It may be something that Sam can clean up, upstream? @yxsamliu

Reply:
It was a temporary fix. We will clean it up later.